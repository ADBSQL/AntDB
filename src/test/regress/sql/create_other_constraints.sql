set grammar to oracle;
--1、非空约束
create table tt(id integer,name varchar2(10) not null);
insert into tt values(1,'');
insert into tt values(2,null);
insert into tt values(3);
drop table tt;
--2、默认值
create table tt(id integer,name varchar2(10) default 'no');
insert into tt values(1);
select * from tt;
drop table tt;
--3、唯一性
create table tt(id integer,name varchar2(10) unique);
insert into tt values(1,null);
insert into tt values(2,null);
insert into tt values(3,'a');
insert into tt values(4,'a');
insert into tt values(5,'');
insert into tt values(6,'');
select * from tt order by id;
drop table tt;
create table tt(id integer,name varchar2(10), constraint unq unique(name));
drop table tt;
create table tt(id integer unique,name varchar2(10) unique);
drop table tt;
create table tt(tid int primary key,name varchar(2) unique);
drop table tt;
create table tt(id integer,name varchar2(10), unique(id,name));
insert into tt values(1,'a');
insert into tt values(1,'b');
insert into tt values(2,'a');
insert into tt values(2,'a');
drop table tt;
--4、检查约束
create table tt(id integer,rate binary_float check(rate>1024) );
insert into tt values(1,500);
insert into tt values(1,1024.1);
select * from tt;
drop table tt;
create table tt(id integer check(id>0), rate binary_float check(rate>1024) );
insert into tt values(0,1024.1);
drop table tt;
create table tt(id integer,rate binary_float check(rate is not null) );
drop table tt;
create table tt(id integer,rate binary_float, constraint chk check(rate > 1000));
insert into tt values(1,500);
insert into tt values(1,1001);
select * from tt;
drop table tt;
create table tt(id integer,rate binary_float, constraint chk check(rate > 1000 and id >100));
insert into tt values(101,1001);
insert into tt values(1,1001);
drop table tt;

