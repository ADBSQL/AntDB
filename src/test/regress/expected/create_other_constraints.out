set grammar to oracle;
--1、非空约束
create table tt(id integer,name varchar2(10) not null);
insert into tt values(1,'');
ERROR:  null value in column "name" violates not-null constraint
DETAIL:  Failing row contains (1, null).
insert into tt values(2,null);
ERROR:  null value in column "name" violates not-null constraint
DETAIL:  Failing row contains (2, null).
insert into tt values(3);
ERROR:  null value in column "name" violates not-null constraint
DETAIL:  Failing row contains (3, null).
drop table tt;
--2、默认值
create table tt(id integer,name varchar2(10) default 'no');
insert into tt values(1);
select * from tt;
 id | name 
----+------
  1 | no
(1 row)

drop table tt;
--3、唯一性
create table tt(id integer,name varchar2(10) unique);
insert into tt values(1,null);
insert into tt values(2,null);
insert into tt values(3,'a');
insert into tt values(4,'a');
ERROR:  duplicate key value violates unique constraint "tt_name_key"
DETAIL:  Key (name)=(a) already exists.
insert into tt values(5,'');
insert into tt values(6,'');
select * from tt order by id;
 id | name 
----+------
  1 | 
  2 | 
  3 | a
  5 | 
  6 | 
(5 rows)

drop table tt;
create table tt(id integer,name varchar2(10), constraint unq unique(name));
drop table tt;
create table tt(id integer unique,name varchar2(10) unique);
ERROR:  Cannot create index whose evaluation cannot be enforced to remote nodes
drop table tt;
ERROR:  table "tt" does not exist
create table tt(tid int primary key,name varchar(2) unique);
ERROR:  Cannot create index whose evaluation cannot be enforced to remote nodes
drop table tt;
ERROR:  table "tt" does not exist
create table tt(id integer,name varchar2(10), unique(id,name));
insert into tt values(1,'a');
insert into tt values(1,'b');
insert into tt values(2,'a');
insert into tt values(2,'a');
ERROR:  duplicate key value violates unique constraint "tt_id_name_key"
DETAIL:  Key (id, name)=(2, a) already exists.
drop table tt;
--4、检查约束
create table tt(id integer,rate binary_float check(rate>1024) );
insert into tt values(1,500);
ERROR:  new row for relation "tt" violates check constraint "tt_rate_check"
DETAIL:  Failing row contains (1, 500).
insert into tt values(1,1024.1);
select * from tt;
 id |  rate  
----+--------
  1 | 1024.1
(1 row)

drop table tt;
create table tt(id integer check(id>0), rate binary_float check(rate>1024) );
insert into tt values(0,1024.1);
ERROR:  new row for relation "tt" violates check constraint "tt_id_check"
DETAIL:  Failing row contains (0, 1024.1).
drop table tt;
create table tt(id integer,rate binary_float check(rate is not null) );
drop table tt;
create table tt(id integer,rate binary_float, constraint chk check(rate > 1000));
insert into tt values(1,500);
ERROR:  new row for relation "tt" violates check constraint "chk"
DETAIL:  Failing row contains (1, 500).
insert into tt values(1,1001);
select * from tt;
 id | rate 
----+------
  1 | 1001
(1 row)

drop table tt;
create table tt(id integer,rate binary_float, constraint chk check(rate > 1000 and id >100));
insert into tt values(101,1001);
insert into tt values(1,1001);
ERROR:  new row for relation "tt" violates check constraint "chk"
DETAIL:  Failing row contains (1, 1001).
drop table tt;
